iocell/src/main/scala/chisel/IOCell.scala

```diff
@@ -0,0 +1,188 @@
+// See LICENSE for license details
+
+package barstools.iocell.chisel
+
+import chisel3._
+import chisel3.util.{Cat, HasBlackBoxResource}
+import chisel3.experimental.{Analog, DataMirror, IO}
+
+class AnalogIOCellBundle extends Bundle {
+  val pad = Analog(1.W)
+  val core = Analog(1.W)
+}
+
+class DigitalGPIOCellBundle extends Bundle {
+  val pad = Analog(1.W)
+  val i = Output(Bool())
+  val ie = Input(Bool())
+  val o = Input(Bool())
+  val oe = Input(Bool())
+}
+
+class DigitalOutIOCellBundle extends Bundle {
+  val pad = Output(Bool())
+  val o = Input(Bool())
+  val oe = Input(Bool())
+}
+
+class DigitalInIOCellBundle extends Bundle {
+  val pad = Input(Bool())
+  val i = Output(Bool())
+  val ie = Input(Bool())
+}
+
+abstract class IOCell extends BlackBox with HasBlackBoxResource
+
+abstract class AnalogIOCell extends IOCell {
+  val io: AnalogIOCellBundle
+}
+
+abstract class DigitalGPIOCell extends IOCell {
+  val io: DigitalGPIOCellBundle
+}
+
+abstract class DigitalInIOCell extends IOCell {
+  val io: DigitalInIOCellBundle
+}
+
+abstract class DigitalOutIOCell extends IOCell {
+  val io: DigitalOutIOCellBundle
+}
+
+class ExampleAnalogIOCell extends AnalogIOCell {
```

Comment:
I'm not sure I like the `Example` prefix. They are more like models of an IO cell, or an RTL only IO cell.

Reply:
Yeah, not super attached to it myself. Is `Generic` better?

Reply:
Sure `Generic` is better.

Reply:
Fixed

Reply:
This is fixed